File: android/capacitor/src/main/java/com/getcapacitor/util/InternalUtils.java
package com.getcapacitor.util;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

public class InternalUtils {

    public static PackageInfo getPackageInfo(PackageManager pm, String packageName) throws PackageManager.NameNotFoundException {
        return InternalUtils.getPackageInfo(pm, packageName, 0);
    }

    public static PackageInfo getPackageInfo(PackageManager pm, String packageName, long flags)
        throws PackageManager.NameNotFoundException {
        return pm.getPackageInfo(packageName, PackageManager.PackageInfoFlags.of(flags));
    }
}

File: android/capacitor/src/main/java/com/getcapacitor/util/InternalUtils.kt
package com.getcapacitor.util

import android.content.pm.PackageInfo
import android.content.pm.PackageManager

object InternalUtils {
    @JvmStatic
    @Throws(PackageManager.NameNotFoundException::class)
    fun getPackageInfo(pm: PackageManager, packageName: String): PackageInfo? = getPackageInfo(pm, packageName, 0)

    @JvmStatic
    @Throws(PackageManager.NameNotFoundException::class)
    fun getPackageInfo(pm: PackageManager, packageName: String, flags: Long): PackageInfo? =
        pm.getPackageInfo(packageName, PackageManager.PackageInfoFlags.of(flags))
}
